### ISSUE TITLE
Potential Issue with maxPriorityFeePerGas estimation

### ISSUE THREAD

**Issue by cloudonshore** [MyCryptoHQ/MyCrypto#4067]
_Github issues are primarily for reporting bugs or getting assistance with development-related items. If you have a question or need support with our product, please email support@mycrypto.com for faster attention._

_If you are reporting an issue, please use the template below so we can reproduce, determine the cause of the issue, and fix it! If you are not reporting an issue, delete this entire thing and type away. Thanks!_

---

### Description of Issue

In the code for fetching the fee history you use the 1st percentile.
https://github.com/MyCryptoHQ/MyCrypto/blob/fb/eip-1559/src/services/ApiService/Gas/eip1559.ts#L10

However, looking at the current fee history on mainnet, fetching the last 5 blocks, there are many `0` value `reward` responses at the 1st percentile. (this is a response to `provider.send('eth_feeHistory', [
    5,
    hexlify(latestBlock.number),
    1,
  ])`)

```
{
   "baseFeePerGas":[
      "0",
      "0",
      "0",
      "0",
      "0",
      "0"
   ],
   "gasUsedRatio":[
      0.9998642662182998,
      0.9987883963929768,
      0.9998072632102536,
      0.998654061598854,
      0.999909687412393
   ],
   "oldestBlock":12947488,
   "reward":[
      [
         "0"
      ],
      [
         "44850000000"
      ],
      [
         "1000000000"
      ],
      [
         "0"
      ],
      [
         "0"
      ]
   ]
}
```
This is most likely due to flashbot transactions that are being inserted directly by the miner. I've found that using the 10th percentile generally filters these out, but is by no means a programmatically dependable method. 

Basically, I think you might potentially be suggesting users use a priority fee of `0` once 1559 goes live on mainnet, because the 1st percentile is often `0` when flashbot transactions are included in the block. I've also seen flashbot transactions submitted with `1` wei for gas price as well, so filtering out for just `0` doesn't seem like a good enough way to catch the issue.

I'm currently developing the 1559 gas fee algorithm for metamask. If you think of other clever ways to get around this issue I would love to jump on a call. My email is sam.walker@consensys.net.

**Comment by FrederikBolding** [comment]
I hadn't seen these values being returned before. But I will definitely look into this. I'll let you know what I find.

**Comment by FrederikBolding** [comment]
@cloudonshore I can indeed see that this is behaviour that happens on mainnet. My initial instinct is that we should probably look at more blocks and then filter out values that are "obviously wrong". But don't have a solution at this time. Let me know if you want to discuss it further.

**Comment by FrederikBolding** [comment]
@cloudonshore There solution I came up in the latest commit consists of multiple things:

- Remove all zero values
- Consider `10` blocks instead of `5`
- Consider `5` percentile instead of `1`
- Slice off values that are significantly lower than the rest of the values (have a 200%+ increase in value) (See: https://github.com/MyCryptoHQ/MyCrypto/blob/fb/eip-1559/src/services/ApiService/Gas/eip1559.ts#L64-L81)

Let me know your thoughts and if you want to discuss any of this more! 😄

**Comment by cloudonshore** [comment]
I think that sounds like a very sane solution. Was implementing something along the same lines.

**Comment by FrederikBolding** [comment]
> I think that sounds like a very sane solution. Was implementing something along the same lines.

Great to hear! Thanks again for reporting!

**Comment by ligi** [comment]
FYI: these zeroes are happening for empty blocks

**Comment by FrederikBolding** [comment]
> FYI: these zeroes are happening for empty blocks

Also that, yes 👍
